### PR TITLE
[Tests][Explicit Module Builds] Fix per-test state in incremental explicitly-built module tests to avoid conflicts on parallel test runs

### PR DESCRIPTION
**Pull request by artemcm**

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
Let's fix-forward instead, with https://github.com/apple/swift-driver/pull/1632.

File: Tests/SwiftDriverTests/IncrementalCompilationTests.swift [hunk 1]
```diff
@@ -19,10 +19,7 @@ import TestUtilities
 
 // MARK: - Instance variables and initialization
 final class IncrementalCompilationTests: XCTestCase {
-
   var tempDir: AbsolutePath = try! AbsolutePath(validating: "/tmp")
```
Comment (adrian-prantl) [inline Tests/SwiftDriverTests/IncrementalCompilationTests.swift:22]: We're not hardcoding `/tmp` to use as a temporary directory, are we? We should have something similar to LLVM's implementation:
https://github.com/llvm/llvm-project/blob/ca33796d54ce6d2c711032b269caf32851c5915a/llvm/lib/Support/Unix/Path.inc#L1452

Comment (artemcm) [reply Tests/SwiftDriverTests/IncrementalCompilationTests.swift:22]: That's the default, but it does get overridden for each test in the `setUp`.

File: Tests/SwiftDriverTests/IncrementalCompilationTests.swift [hunk 2]
```diff
@@ -90,15 +109,38 @@ final class IncrementalCompilationTests: XCTestCase {
 
   override func setUp() {
     self.tempDir = try! withTemporaryDirectory(removeTreeOnDeinit: false) {$0}
```
Comment (adrian-prantl) [inline Tests/SwiftDriverTests/IncrementalCompilationTests.swift:111]: I see!

**Review by adrian-prantl — APPROVED** [review]

**Pull request merged**